cmd/cli/kubectl-kyverno/commands/test/test.go

```diff
@@ -98,6 +98,29 @@ func runTest(out io.Writer, testCase test.TestCase, registryAccess bool, auditWa
 	} else {
 		fmt.Fprintln(out, "  Applying", len(policies)+len(validatingAdmissionPolicies), pluralize.Pluralize(len(policies)+len(validatingAdmissionPolicies), "policy", "policies"), "to", len(uniques), pluralize.Pluralize(len(uniques), "resource", "resources"), "...")
 	}
+	//For veriifying all policies and rules are mentioned in test are also in policy
+        for _, res := range testCase.Test.Results {
+		ruleSame := false
+		policySame := false
+		for _, policy := range policies {
+			if res.Policy == policy.GetName() || res.Policy == "default/"+policy.GetName() {
+				policySame = true
+				for _, rule := range autogen.ComputeRules(policy) {
+					if res.Rule == rule.Name {
+						ruleSame = true
+						break
+					}
+				}
+			}
+		}
+		if !policySame {
+			return nil, fmt.Errorf("the policy(%v) name not found", res.Policy)
+		}
+		if !ruleSame {
+			return nil, fmt.Errorf("the rule(%v) name not found", res.Rule)
```

Comment:
This isn't a good error message. The rule name doesn't need to be in parenthesis and it should tell the user in which policy the rule is expected to be found.

```
The rule foo cannot be found in the policy named foobar.
```

Reply:
fixed

Reply:
fixed